plugins/api/rate-limit/src/main/java/org/apache/cloudstack/ratelimit/ApiRateLimitServiceImpl.java

```diff
@@ -139,24 +139,33 @@ public boolean resetApiLimit(Long accountId) {
         return true;
     }
 
+    @Override
+    public boolean isEnabled() {
+        return enabled;
+    }
+
     @Override
     public boolean checkAccess(User user, String apiCommandName) throws PermissionDeniedException {
         // check if api rate limiting is enabled or not
         if (!enabled) {
             return true;
         }
-        Long accountId = user.getAccountId();
-        Account account = _accountService.getAccount(accountId);
+        Account account = _accountService.getAccount(user.getAccountId());
+        return checkAccessWithoutEnabledCheck(account, user, apiCommandName, null);
+    }
+
+    @Override
+    public boolean checkAccessWithoutEnabledCheck(Account account, User user, String apiCommandName, Role role) throws PermissionDeniedException {
```

Comment:
i'd say, pass accountId instead of account and leave the accountName out of the trace message.

Reply:
Can do but don't see particular benefit of that and will have to tinker the code (trace bit) that's working fine.